### ISSUE TITLE
Component Repeated Mount

### ISSUE THREAD

**Issue by nodece** [react-navigation/react-navigation#3203]
# I think this is 🐛 
## scene
I reset route go Main screen and set navigation params in Main screen。 Main screen should only once mounted，but Main screen Repeated mount。

## Code
Guide Screen
```
let routeName = 'Main'
const resetAction = NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName })
      ]
    })
this.props.navigation.dispatch(resetAction)
```
Main Screen
```
componentDidMount() {
    const { navigate } = this.props.navigation
    this.props.navigation.setParams({
        userLogout: ()=>{},
        triggerPicker:  ()=>{}
    })
}
```

**Comment by kelset** [comment]
Closing because you haven't followed the issue template; please open a new one that follows the template.
Without that, it's nearly impossible to understand & investigate the issue. I suggest you also try (in the new issue) to answer/explain in a way that fits these questions:

1. have you tried updating to the latest release?
1. does it happen only on Android/iOS?
1. does it happen on both Debug and Release modes?
1. does it happen with Debugger On or Off? Both?
1. Can you provide a GIF or a repo or a snack to reproduce this issue?

Moreover, I think it's actually a duplicate of #2599

**Comment by Sommerfeld** [comment]
I do have the same issue in beta23 and do not think it is a duplicate of the issue #2599 . My priority for now is to find a workaround or fix for that before I maybe create a sample app and create a detailed issue.

What I am doing is simply reset my MainStackNaivgator with subNavigators and call setParams in one of the screens of a SubTabNavigator. This leads to continuously mounting the screen of the SubTabNavigator.

**Comment by Sommerfeld** [comment]
#2702 seems to be related.

@nodece calling setParams in InteractionManager.runAfterInteractions() as mentioned in this issue might "fix" it for you. It does for me.

**Comment by nodece** [comment]
@Sommerfeld  Thanks， It is worked，Calling setParams in InteractionManager.runAfterInteractions()
